controller.go

```diff
@@ -196,15 +196,79 @@ func New(cfgOptions ...config.Option) (NetworkController, error) {
 		}
 	}
 
-	c.sandboxCleanup()
+	c.sandboxCleanup(oldRunningContainers)
+	if err := c.restoreSandbox(oldRunningContainers); err != nil {
+		log.Errorf("failed to restore sandbox")
+	}
+
 	c.cleanupLocalEndpoints()
 	c.networkCleanup()
 
 	if err := c.startExternalKeyListener(); err != nil {
-		return nil, err
+		return nil, nil, err
+	}
+	restored := make(map[string]interface{})
+	for _, sb := range c.sandboxes {
+		restored[sb.ContainerID()] = true
 	}
 
-	return c, nil
+	return c, restored, nil
+}
+
+func (c *controller) restoreSandbox(sbids map[string]interface{}) error {
+	for id, sb := range c.sandboxes {
+		log.Infof("restore sandbox %s of container %s", sb.ID(), sb.ContainerID())
+		option, ok := sbids[sb.ContainerID()].([]SandboxOption)
+		if !ok {
+			log.Errorf("failed to restore sandbox: no restore options passed from daemon")
+			delete(c.sandboxes, id)
+			continue
+		}
+		err := sb.restoreSandbox(option)
+		if err != nil {
+			log.Errorf("failed to restore sandbox %s", sb.ID())
```

Comment:
Please also print the error.

Across the diffs, pls follow the convention where Log message start with upper case and error msg with lower case.